src/lib.rs

```diff
@@ -1,17 +1,46 @@
-mod connect;
-mod sc;
-mod replica;
-mod metadata;
 mod admin;
+mod connect;
+mod consumer;
+mod producer;
+mod fluvio;
 
-
-use shared::SharedScClient;
+use shared::SharedFluvio;
 
 mod shared {
 
     use std::sync::Arc;
-    use flv_client::ScClient;
+    use fluvio::Fluvio;
     use flv_future_aio::sync::RwLock;
 
-    pub type SharedScClient = Arc<RwLock<ScClient>>;
-}
\ No newline at end of file
+    pub type SharedFluvio = Arc<RwLock<Fluvio>>;
```

Comment:
As of the Rust API cleanup, I believe all of the methods on the `Fluvio` object take `&self` rather than `&mut self`, so it should be possible to make `SharedFluvio` into just an `Arc<Fluvio>` and avoid the `RwLock`. I think we should make it a design pattern to make this a rule for the Rust API. That is, we should encapsulate any locking that needs to have with interior-mutability patterns so that external clients like node don't need to worry about it.

Reply:
Actually,  Fluvio itself it sharable since all it's parts are sharable.   I think only thing missing is `Clone` derive.